src/components/alert/index.jsx

```diff
@@ -1,17 +1,37 @@
+import { useEffect } from 'react'
+import { useAlertMessage } from '../../hooks'
+
 const variants = {
    success: 'bg-green-500',
    error: 'bg-red-500',
    info: 'bg-blue-500'
 }
 
 export const Alert = ({ message, variant }) => {
+   const { setShowAlert, showAlert, displayAlert } = useAlertMessage()
```

Comment:
I think we don't need to use the custom hook inside of the alert component, a local state is enough to handle if the component is displayed or not